java/memory/src/main/java/org/apache/arrow/memory/BoundsChecking.java

```diff
@@ -26,7 +26,7 @@
     boolean isAssertEnabled = false;
     assert isAssertEnabled = true;
     BOUNDS_CHECKING_ENABLED = isAssertEnabled ||
-      !"true".equals(System.getProperty("drill.enable_unsafe_memory_access"));
+      "true".equals(System.getProperty("arrow.check_unsafe_memory_access"));
```

Comment:
You might need to also check the old property for backwards compatibility (possibly raising a deprecation warning)